jupyter_server/services/sessions/sessionmanager.py

```diff
@@ -69,48 +68,39 @@ def session_exists(self, path):
             # row, thereby terminating the session.  This can be done via a call to
             # row_to_model that tolerates that condition.  If row_to_model returns None,
             # we'll return false, since, at that point, the session doesn't exist anyway.
-            model = yield maybe_future(self.row_to_model(row, tolerate_culled=True))
+            model = await self.row_to_model(row, tolerate_culled=True)
             if model is not None:
                 exists = True
-        raise gen.Return(exists)
+        return exists
 
     def new_session_id(self):
         "Create a uuid for a new session"
         return unicode_type(uuid.uuid4())
 
-    @gen.coroutine
-    def create_session(self, path=None, name=None, type=None, kernel_name=None, kernel_id=None):
+    async def create_session(self, path=None, name=None, type=None, kernel_name=None, kernel_id=None):
         """Creates a session and returns its model"""
         session_id = self.new_session_id()
         if kernel_id is not None and kernel_id in self.kernel_manager:
             pass
         else:
-            kernel_id = yield self.start_kernel_for_session(session_id, path, name, type, kernel_name)
-        result = yield maybe_future(
-            self.save_session(session_id, path=path, name=name, type=type, kernel_id=kernel_id)
-        )
-        # py2-compat
-        raise gen.Return(result)
+            kernel_id = await self.start_kernel_for_session(session_id, path, name, type, kernel_name)
+        result = await self.save_session(session_id, path=path, name=name, type=type, kernel_id=kernel_id)
+        return result
 
-    @gen.coroutine
-    def start_kernel_for_session(self, session_id, path, name, type, kernel_name):
+    async def start_kernel_for_session(self, session_id, path, name, type, kernel_name):
         """Start a new kernel for a given session."""
         # allow contents manager to specify kernels cwd
         kernel_path = self.contents_manager.get_kernel_path(path=path)
-        kernel_id = yield maybe_future(
-            self.kernel_manager.start_kernel(path=kernel_path, kernel_name=kernel_name)
-        )
-        # py2-compat
-        raise gen.Return(kernel_id)
+        kernel_id = await ensure_async(self.kernel_manager.start_kernel(path=kernel_path, kernel_name=kernel_name))
```

Comment:
I don't think we need the `ensure_async()` wrapper here.

Reply:
Good point, and this involves some changes in the tests too.